### PR TITLE
gradleify project

### PR DESCRIPTION
**Pull request by loki666**
first step before publishing on jcenter #30

### CODE REVIEW

**Comment by philburk** [comment]
This looks really good. The original development was using Android.mk and the project was never fully ported to Android Studio.
Can you please describe how you did this work?
Did you hand build the files like gradlew or did Android Studio generate them?
Then I will know how closely I need to review them.

**Comment by loki666** [comment]
basically, I recreated the whole project in Android Studio, module by module
then moved the .git folder of this repository in my project
since I didn't modify the content of the files, git was smart enough to figure move files like a champ!

**Comment by philburk** [comment]
Yes, git is amazing with recognizing moves. Sounds like a good procedure.

Did you hand build any of the files like gradlew or did Android Studio generate them?

I will check with the Android Studio folks next week and then do the merge. Thanks for doing this work.

**Comment by loki666** [comment]
I only edited the build.gradle of the modules to add the dependency to
miditools and set the versionCode and versionName.

The rest was created by Android Studio when I created the project and the
modules.

On Dec 11, 2016 02:24, "Phil Burk" <notifications@github.com> wrote:

Yes, git is amazing with recognizing moves. Sounds like a good procedure.

Did you hand build any of the files like gradlew or did Android Studio
generate them?

I will check with the Android Studio folks next week and then do the merge.
Thanks for doing this work.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/philburk/android-midisuite/pull/31#issuecomment-266254263>,
or mute the thread

File: midikeyboard/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,12 +1,9 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.mobileer.midikeyboard"
-        android:versionCode="4"
-        android:versionName="@string/versionName">
```
Comment (philburk) [inline midikeyboard/src/main/AndroidManifest.xml]: We lost a feature. The versionName was grabbed from strings.xml. But I suppose gradle may not support this.

File: miditools/src/main/java/com/mobileer/miditools/MidiDispatcher.java [hunk 2]
```diff
@@ -23,11 +23,11 @@
 import java.util.concurrent.CopyOnWriteArrayList;
 
 /**
- * Utility class for dispatching MIDI data to a list of {@link android.media.midi.MidiReceiver}s.
```
Comment (philburk) [inline miditools/src/main/java/com/mobileer/miditools/MidiDispatcher.java]: Losing the full class name might be a problem.

File: midikeyboard/src/main/AndroidManifest.xml (hunk 1 above)
Comment (loki666) [inline midikeyboard/src/main/AndroidManifest.xml]: Well, Gradle can't reference values from strings.xml.
But Gradle can inject string resources, so we can do the reverse.
Having Gradle generate the versionName value and a R.string.version_name for use in the application.
I'll update the pull request.

File: miditools/src/main/java/com/mobileer/miditools/MidiDispatcher.java (hunk 2 above)
Comment (loki666) [inline miditools/src/main/java/com/mobileer/miditools/MidiDispatcher.java]: javadoc is smart enough to use the full class that match the import

**Comment by loki666** [comment]
should be ok

**Comment by philburk** [comment]
Thanks for doing this work. It needed to be done.

**Pull request merged**